packages/ui/src/components/editor/properties/variant/index.tsx

```diff
@@ -0,0 +1,325 @@
+import { Button } from '@mui/material'
+import BooleanInput from '../../input/Boolean'
+import InputGroup from '../../input/Group'
+import ModelInput from '../../input/Model'
+import NumericInput from '../../input/Numeric'
+import SelectInput from '../../input/Select'
+import NodeEditor from '../nodeEditor'
+
+const buildBudgetVariantMetadata = (
+  level: VariantLevel,
+  signal: AbortSignal,
+  callback: (maxTextureSize: number, vertexCount: number) => void
+) => {
+  const src = level.src
+  const resources = getState(ResourceState).resources
+  if (resources[src] && resources[src].status == ResourceStatus.Loaded) {
+    const metadata = getState(ResourceState).resources[src].metadata as { verts: number; textureWidths: number[] }
+    const maxTextureSize = metadata.textureWidths ? Math.max(...metadata.textureWidths) : 0
+    const verts = metadata.verts
+    callback(maxTextureSize, verts)
+    return
+  }
+
+  loadResource(
+    src,
+    ResourceType.GLTF,
+    UndefinedEntity,
+    () => {
+      const metadata = getState(ResourceState).resources[src].metadata as { verts: number; textureWidths: number[] }
+      const maxTextureSize = metadata.textureWidths ? Math.max(...metadata.textureWidths) : 0
+      const verts = metadata.verts
+      callback(maxTextureSize, verts)
+      ResourceManager.unload(src, UndefinedEntity)
+    },
+    () => {},
+    (error) => {
+      console.warn(
+        `VariantNodeEditor:buildBudgetVariantMetadata: error loading ${src} to build variant metadata`,
+        error
+      )
+      callback(0, 0)
+    },
+    signal
+  )
+}
+
+export const VariantNodeEditor: EditorComponentType = (props: { entity: Entity }) => {
+  const { t } = useTranslation()
+  const entity = props.entity
+  const variantComponent = useComponent(entity, VariantComponent)
+  const previewIndex = useHookstate(0)
+
+  const setPreview = (index: number) => {
+    previewIndex.set(index)
+    const modelComponent = getOptionalMutableComponent(entity, ModelComponent)
+    if (!modelComponent) return
+    modelComponent.src.set(variantComponent.levels[index].src.value)
+  }
+
+  const updateLevelsForBudget = () => {
+    const levels = variantComponent.levels.value
+    const srcSet = new Set<string>()
+    for (const level of levels) {
+      srcSet.add(level.src)
+    }
+
+    if (srcSet.size !== levels.length) {
+      const newLevels: VariantLevel[] = [...srcSet].map((src) => {
+        return {
+          src: src,
+          metadata: {}
+        }
+      })
+      commitProperties(VariantComponent, { levels: newLevels, heuristic: Heuristic.BUDGET }, [entity])
+    } else {
+      commitProperty(VariantComponent, 'heuristic')(Heuristic.BUDGET)
+    }
+  }
+
+  return (
+    <NodeEditor
+      name={t('editor:properties.variant.name')}
+      description={t('editor:properties.variant.description')}
+      icon={<VariantNodeEditor.iconComponent />}
+      {...props}
+    >
+      <div className="m-4 rounded-lg bg-gray-800 p-4">
```

Comment:
![image](https://github.com/EtherealEngine/etherealengine/assets/55396651/71f98aa3-e2dc-4667-b336-cd8605f024aa)

let's make this a darker shade of gray instead of blue

Reply:
fixed it